deposit/src/test/java/edu/unc/lib/deposit/fcrepo4/StaffOnlyPermissionJobTest.java

```diff
@@ -0,0 +1,165 @@
+import edu.unc.lib.dl.rdf.Cdr;
+import edu.unc.lib.dl.rdf.CdrDeposit;
+import org.apache.commons.io.FileUtils;
+import org.apache.jena.rdf.model.Bag;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.Resource;
+import org.apache.jena.vocabulary.RDF;
+import org.junit.Before;
+import org.junit.Test;
+
+import edu.unc.lib.dl.util.RedisWorkerConstants.DepositField;
+import edu.unc.lib.dl.fedora.PID;
+
+/**
+ * @author lfarrell
+ */
+public class StaffOnlyPermissionJobTest extends AbstractDepositJobTest {
+    private final static String LOC1_ID = "loc1";
+    private final static String LOC2_ID = "loc2";
+    private final static String FILE_CONTENT1 = "Some content";
+    private final static String SOURCE_ID = "source1";
+
+    private StaffOnlyPermissionJob job;
+    private Model model;
+    private StorageLocationManager locationManager;
+    private Bag depBag;
+    private StorageLocationTestHelper locTestHelper;
+    private Path loc1Path;
+    private Path loc2Path;
+    private Path candidatePath;
+    private Path sourcePath;
+
+    @Before
+    public void setup() throws Exception {
+        loc1Path = tmpFolder.newFolder("loc1").toPath();
+        loc2Path = tmpFolder.newFolder("loc2").toPath();
+
+        locTestHelper = new StorageLocationTestHelper();
+        locTestHelper.addStorageLocation(LOC1_ID, "Location 1", loc1Path.toString());
+        locTestHelper.addStorageLocation(LOC2_ID, "Location 2", loc2Path.toString());
+
+        sourcePath = tmpFolder.newFolder(SOURCE_ID).toPath();
+        locationManager = locTestHelper.createLocationManager(null);
+        candidatePath = Files.createDirectories(sourcePath.resolve(depositUUID));
+
+        job = new StaffOnlyPermissionJob();
+
+        job.setJobUUID(jobUUID);
+        job.setDepositUUID(depositUUID);
+        job.setDepositDirectory(depositDir);
+        setField(job, "locationManager", locationManager);
+        setField(job, "dataset", dataset);
+        setField(job, "depositsDirectory", depositsDirectory);
+        setField(job, "depositStatusFactory", depositStatusFactory);
+        setField(job, "jobStatusFactory", jobStatusFactory);
+        job.init();
+
+        model = job.getWritableModel();
+        depBag = model.createBag(depositPid.getRepositoryPath());
+        depBag.addProperty(Cdr.storageLocation, LOC1_ID);
+    }
+
+    @Test
+    public void testStaffOnly() throws Exception {
+        Map<String, String> depositStatus = new HashMap<>();
+        depositStatus.put(DepositField.staffOnly.name(), "true");
+        when(depositStatusFactory.get(eq(depositUUID))).thenReturn(depositStatus);
+
+        PID objPid = makePid();
+        Bag objBag = model.createBag(objPid.getRepositoryPath());
+        objBag.addProperty(RDF.type, Cdr.Folder);
+
+        Resource fileResc = addFileObject(objBag, FILE_CONTENT1);
+        objBag.addProperty(Cdr.primaryObject, fileResc);
```

Comment:
i know this is test data, but a folder isn't allow to have a primaryObject. You could remove this property, or make objBag into a work. Also, the child could be any type, it being a file with a storage location isn't necessary for this test.